src/KernelApplication.cs

```diff
@@ -259,21 +306,43 @@ public int InstallKernelSpec(bool develop, LogLevel logLevel, string prefix = nu
 
             // Make a temporary directory to hold the kernel spec.
             var tempKernelSpecDir = Path.Combine(Path.GetTempPath(), Path.GetRandomFileName());
+            var filesToDelete = new List<string>();
             var jsonPath = Path.Combine(tempKernelSpecDir, "kernel.json");
             Directory.CreateDirectory(tempKernelSpecDir);
             File.WriteAllText(jsonPath, JsonConvert.SerializeObject(kernelSpec));
+            filesToDelete.Add(jsonPath);
             kernelSpecDir = tempKernelSpecDir;
 
+            // Add any additional files we may need.
+            if (additionalFiles != null) {
+                foreach (var (fileName, streamAction) in additionalFiles)
+                {
+                    var dest = Path.Combine(tempKernelSpecDir, fileName);
+                    var sourceStream = streamAction();
+                    using (var destStream = File.OpenWrite(dest))
```

Comment:
what happens if the dest path is invalid?

Reply:
I believe that the install should fail in that case, as there's a needed filethat can't be correctly added to the kernelspec directory. We could fail with a better exception, though.